api_design/DDS-974-enforce_upload_chunk_size.md

```diff
@@ -0,0 +1,54 @@
+# DDS-974 Enforce Upload Chunk Size
+
+## Deployment View
+
+N/A
+
+## Logical View
+
+A critical issue was recently exposed with the chunked upload capability.  The  storage provider (Swift), is configured with a set maximum number of segments (i.e. Data Service chunks), beyond which it cannot coalesce a large file.  To remedy this situation, the API will be extended to enforce a minumum chunk size, based on the overall file size and max segments setting.
+
+#### Summary of impacted APIs
+
+|Endpoint |Description |
+|---|---|
+| `POST /projects/{id}/uploads` | Intitate a chunked upload. |
+| `PUT /uploads/{id}/chunks` | Generate and return a pre-signed upload URL for a chunk.  |
+
+#### API Specification
+This section defines the proposed API interface extensions.
+
+##### Intitate a chunked upload
+`POST /projects/{id}/uploads`
```

Comment:
should we return an error if the user attempts to initiate an upload of a file that is too large (max_segments * max_chunk_upload_size)?

Reply:
yes - and do we want to extend storage provider endpoint to return the max segments, min size, max size, that we support at API level?  i assume you would keep these settings at this level - may be good to return them for informational purposes as well.

Reply:
If you can document these proposed changes to the StorageProvider in the api_design doc, that should set it up for us to start implementation.